comp/otelcol/otlp/components/pipeline/provider/provider.go

```diff
@@ -17,15 +18,28 @@ import (
 	"go.opentelemetry.io/collector/confmap/provider/httpsprovider"
 	"go.opentelemetry.io/collector/confmap/provider/yamlprovider"
 	"go.opentelemetry.io/collector/otelcol"
+	"gopkg.in/yaml.v3"
 )
 
+type ExtendedConfigProvider interface {
+	otelcol.ConfigProvider
+	GetProvidedConf() []byte
```

Comment:
Why is it byte array instead of string ?

Reply:
no reason other than the call to yaml.Marshal returns a `[]byte`, so I stored it in this way. I can apply the `string()` and store it as a string ?